src/SearchNamespace.js

```diff
@@ -0,0 +1,155 @@
+        elem.querySelector(SearchNamespace.selectors.cardSubTitle).textContent,
+        // DEV: Remove trailing query params from image with path such as
+        //      https://lh3.googleusercontent.com/4Yht2ETGQNme6QgQi-imsOK788OEHEhldqgBjeR8hWi8YsUMbn_AY0c5COHB4wK5C3Hjiw-y3Q=w220-c-h220-e100
+        elem.querySelector('img').src.replace('=w220-c-h220-e100', '')
+      ));
+    });
+
+    const artistElems = document.querySelectorAll(SearchNamespace.selectors.artistResults);
+    const artists = [];
+
+    Array.prototype.forEach.call(artistElems, (elem) => {
+      let image = elem.querySelector('img');
+      if (image) {
+        // DEV: Remove trailing query params from image with path such as
+        //      https://lh3.googleusercontent.com/4Yht2ETGQNme6QgQi-imsOK788OEHEhldqgBjeR8hWi8YsUMbn_AY0c5COHB4wK5C3Hjiw-y3Q=w190-c-h190-e100
+        image = image.src.replace('=w190-c-h190-e100', '');
+      } else {
+        image = null;
+      }
+      artists.push(new Artist(
+        elem.getAttribute('data-id'),
+        elem.querySelector(SearchNamespace.selectors.cardTitle).textContent,
+        image
+      ));
+    });
+
+    const trackElems = document.querySelectorAll(SearchNamespace.selectors.trackResults);
+    const tracks = [];
+    Array.prototype.forEach.call(trackElems, (elem, index) => {
+      const durationParts = elem.querySelector(SearchNamespace.selectors.trackColumns.duration).textContent.trim().split(':');
+      tracks.push(new Track({
+        id: elem.getAttribute('data-id'),
+        title: this._text(elem.querySelector(SearchNamespace.selectors.trackColumns.title), 'Unknown Title'),
+        albumArt: elem.querySelector(`${SearchNamespace.selectors.trackColumns.title} img`).src.replace('=s60-e100-c', ''),
+        artist: this._text(elem.querySelector(SearchNamespace.selectors.trackColumns.artist), 'Unknown Artist'),
+        album: this._text(elem.querySelector(SearchNamespace.selectors.trackColumns.album), 'Unknown Album'),
+        index,
+        duration: parseInt(durationParts[0], 10) * 60 + parseInt(durationParts[1], 10),
+        playCount: parseInt(this._text(elem.querySelector(SearchNamespace.selectors.trackColumns.playCount), '0'), 10),
+      }));
+    });
+
+    return {
+      albums,
+      artists,
+      tracks,
+    };
+  }
+
+  isSearching() {
+    return /^#\/sr\//g.test(window.location.hash);
+  }
+
+  playResult(resultObject) {
+    const trackPlay = document.querySelector(`[data-id="${resultObject.id}"] ${SearchNamespace.selectors.playButton}`);
+    const otherPlay = document.querySelector(`[data-id="${resultObject.id}"] ${SearchNamespace.selectors.cardPlayButton}`);
+    if (!trackPlay && !otherPlay) {
+      throw new Error('Failed to play result, it must not be in this search');
+    }
+    (trackPlay || otherPlay).click();
+  }
+
+  performSearch(text) {
+    window.location.hash = `/sr/${escape(text.replace(/ /g, '+'))}`;
```

Comment:
If this is JS's escape (which I think it is), then we should be using `encodeURIComponent` which is proper for URLs:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent